### PR TITLE
Some cleanup in LoadEventNexus

### LINKED ISSUE #20419: Cleanup LoadEventNexus
Cleanup/refactor relevant parts of `LoadEventNexus` as a preparation for follow-up work related to #20344.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Extract some code from `LoadEventNexus` and move it to separate files/classes. This is a preparation for having an alternative loader for event data.

Note that dependency issues have not been cleaned up, such that, e.g., `LoadBankFromDiskTask` is still referencing internal variables in `LoadEventNexus` (and `DefaultEventLoader` to which some of the code has been migrated). I think this is beyond the scope of this PR an can be dealt with later if necessary.

**To test:**

Code review might become easier when reviewing each commit individually instead of the full set of changes. Each commit is mostly just moving around code.

Fixes #20419. 
Internal change -> no release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by LamarMoore — APPROVED** [review]
Changes look good to me

**Comment by peterfpeterson** [comment]
@wdzhou should know about this

**Review by OwenArnold — COMMENTED** [review]
This has been closed as I was editing for review.

File: Framework/DataHandling/src/LoadEventNexus.cpp [hunk 1]
```diff
@@ -1507,96 +850,13 @@ void LoadEventNexus::loadEvents(API::Progress *const prog,
...
-          this, bankNames[i], classType, bankNumEvents[i], oldNeXusFileNames,
-          prog2.get(), diskIOMutex, scheduler, periodLogVec));
-  }
-  // Start and end all threads
-  pool.joinAll();
-  diskIOMutex.reset();
+  bool precount = getProperty("Precount");
+  int chunk = getProperty("ChunkNumber");
+  int totalChunks = getProperty("TotalChunks");
+  DefaultEventLoader::load(this, *m_ws, haveWeights, event_id_is_spec,
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply Framework/DataHandling/src/LoadEventNexus.cpp]: @OwenArnold Cannot think of a better name. The strategy is to support all kinds of files/special cases and files with potentially bad data. Do you have a good suggestion?

**Comment by SimonHeybrock** [comment]
@OwenArnold I will your comments in some other PR going into the feature branch.

File: Framework/DataHandling/src/LoadEventNexus.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (OwenArnold) [reply Framework/DataHandling/src/LoadEventNexus.cpp]: Maybe this suggests that some further refactoring is needed later. Leave it as `Default` for now. I don't have a better suggestion.

File: Framework/DataHandling/src/DefaultEventLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+#include "MantidKernel/make_unique.h"
+
+using namespace Mantid::Kernel;
+
+namespace Mantid {
+namespace DataHandling {
+
+void DefaultEventLoader::load(LoadEventNexus *alg, EventWorkspaceCollection &ws,
+                              bool haveWeights, bool event_id_is_spec,
+                              std::vector<std::string> bankNames,
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply Framework/DataHandling/src/DefaultEventLoader.cpp]: Turns out those were passed by value on purpose because they get modified.

**Pull request merged**